src/postgraphile/postgraphile.ts

```diff
@@ -114,41 +114,36 @@ export default function postgraphile(
   // must process them with `pluginHook` before we can rely on them.
   let incomingOptions: PostGraphileOptions;
 
-  // If the second argument is undefined, use defaults for both `schema` and
-  // `incomingOptions`.
-  if (typeof schemaOrOptions === 'undefined') {
-    schema = 'public';
-    incomingOptions = {};
-  }
   // If the second argument is a string or array, it is the schemas so set the
   // `schema` value and try to use the third argument (or a default) for
   // `incomingOptions`.
-  else if (typeof schemaOrOptions === 'string' || Array.isArray(schemaOrOptions)) {
+  if (typeof schemaOrOptions === 'string' || Array.isArray(schemaOrOptions)) {
     schema = schemaOrOptions;
     incomingOptions = maybeOptions || {};
   }
-  // Otherwise the second argument is the incomingOptions so set `schema` to the
+  // If the second argument is the incomingOptions so set `schema` to the
   // default and `incomingOptions` to the second argument.
-  else {
+  else if (typeof schemaOrOptions !== 'undefined') {
     schema = 'public';
     incomingOptions = schemaOrOptions;
   }
+  // Otherwise the second argument is undefined, use defaults for both `schema` and
+  // `incomingOptions`.
+  else {
+    if (arguments.length > 1) {
+      throw new Error('The second argument to postgraphile was `undefined`... did you mean to set a schema?');
+    }
+    schema = 'public';
+    incomingOptions = {};
+  }
+
+  if (typeof poolOrConfig === 'undefined' && arguments.length > 1) {
```

Comment:
```suggestion
  if (typeof poolOrConfig === 'undefined' && arguments.length >= 1) {
```

Should this be `>=` - i.e. if they passed undefined?